src/frontends/paddle/src/op/reverse.cpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "default_opset.hpp"
+#include "openvino/opsets/opset1.hpp"
+#include "openvino/frontend/paddle/node_context.hpp"
+
+namespace ov {
+namespace frontend {
+namespace paddle {
+namespace op {
+
+using namespace default_opset;
+
+NamedOutputs reverse(const NodeContext& node) {
+    auto x = node.get_input("X");
+    // axis is a vector
+    auto axis = node.get_attribute<std::vector<int32_t>>("axis");
+    const auto rank = x.get_partial_shape().rank();
+
+    if (rank == 1) {
+        auto axis_node = default_opset::Constant::create(ngraph::element::i32, Shape{1}, axis);
+        auto reverse_op = std::make_shared<ov::opset1::Reverse>(x, axis_node, opset1::Reverse::Mode::INDEX);
+        auto out = node.default_single_output_mapping({reverse_op}, {"Out"});
+        return out;
+    } else {
+        // get value from axis
+        auto seq_axis = axis.at(0);
+        // batch_axis
+        int64_t batch_axis = !seq_axis;
+        //
+        auto shape = std::make_shared<ShapeOf>(x);
+        auto one = std::make_shared<Constant>(element::i64, Shape{1}, 1);
+        auto gather_batch = std::make_shared<Gather>(shape,
+                                                     std::make_shared<Constant>(element::i64, Shape{1}, batch_axis),
+                                                     std::make_shared<Constant>(element::i64, Shape{1}, 0));
+
+        auto gather_seq = std::make_shared<Gather>(shape,
+                                                   std::make_shared<Constant>(element::i64, Shape{1}, batch_axis),
+                                                   std::make_shared<Constant>(element::i64, Shape{1}, 0));
+        auto broadcast = std::make_shared<Broadcast>(one, gather_batch);
+        auto seq_lengths = std::make_shared<Multiply>(broadcast, gather_seq);
```

Comment:
Seems gather_batch and gather_seq are the same thing. Could you please add comments to the code to explain the principle of the code?

Reply:
Frankly, cannot understand either. Sounds OV reverse could cover Paddle reverse.

Reply:
@ceciliapeng2011 @luo-cheng2021 rewrite and test the reverse codes.